src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -49,7 +51,23 @@ internal NumericValueSet(ImmutableArray<(T first, T last)> intervals)
                 _intervals = intervals;
             }
 
-            bool IValueSet.IsEmpty => _intervals.Length == 0;
+            public bool IsEmpty => _intervals.Length == 0;
+
+            ConstantValue IValueSet.Sample
+            {
+                get
+                {
+                    // Prefer a value near zero.
+                    var tc = default(TTC);
+                    var gz = NumericValueSetFactory<T, TTC>.Instance.Related(BinaryOperatorKind.GreaterThanOrEqual, tc.Zero);
+                    var t = (NumericValueSet<T, TTC>)this.Intersect(gz);
+                    if (!t.IsEmpty)
+                        return tc.ToConstantValue(t._intervals[0].first);
+                    t = (NumericValueSet<T, TTC>)this.Intersect(gz.Complement());
```

Comment:
Is there any difference between `this` and `this.Intersect(gz.Complement())`, if `this.Intersect(gz)` is empty?

Reply:
No. Good point.

---
In reply to: [441171418](https://github.com/dotnet/roslyn/pull/44702#discussion_r441171418) [](ancestors = 441171418)